### PR TITLE
split actual winrm "stderr" (xml) into a filtered "stderr" stream

### PR DESCRIPTION
**Pull request by ahgittin**
… and an exhaustive "xml" error stream

needs testing and want to make other streams available but for early review

### CODE REVIEW

**Comment by ahgittin** [comment]
This makes three output streams for winrm commands:

* stdout = usual
* xmlout = the full xml we get from powershell, pretty-printed (using @duncangrant 's recent work)
* stderr = xmlout de-xml-ified filtered for Error stream messages

For example, as per the new live test `testDifferentLogLevels` given the input script:

```
$DebugPreference = "Continue"
$VerbosePreference = "Continue"
Write-Host "Log in Host Stream"
Write-Output "Log in Success Stream"
Write-Error "Log in Error Stream" 
Write-Warning "Log in Warning Stream"
Write-Verbose "Log in Verbose Stream"
Write-Debug "Log in Debug Stream" 
```

`stdout` has, both from winrm and in our ui:

```
Log in Host Stream
Log in Success Stream
```

the actual `stderr` stream as returned from winrm has XML, which is not as helpful as people might wish stderr to be, and is not what the MS docs say(https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_output_streams?view=powershell-7.1); so we pretty format this and return it in an `xmlout` stream for the ui:

```
#< CLIXML
<Objs Version="1.1.0.1" xmlns="http://schemas.microsoft.com/powershell/2004/04">
	<S S="Error">$DebugPreference = "Continue"_x000D__x000A_</S>
	<S S="Error">$VerbosePreference = "Continue"_x000D__x000A_</S>
	<S S="Error">Write-Host "Log in Host Stream"_x000D__x000A_</S>
	<S S="Error">Write-Output "Log in Success Stream"_x000D__x000A_</S>
	<S S="Error">Write-Error "Log in Error Stream" _x000D__x000A_</S>
	<S S="Error">Write-Warning "Log in Warning Stream"_x000D__x000A_</S>
	<S S="Error">Write-Verbose "Log in Verbose Stream"_x000D__x000A_</S>
	<S S="Error">Write-Debug "Log in Debug Stream" _x000D__x000A_</S>
	<S S="Error"> : Log in Error Stream_x000D__x000A_</S>
	<S S="Error">    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorExcep _x000D__x000A_</S>
	<S S="Error">   tion_x000D__x000A_</S>
	<S S="Error">    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorExceptio _x000D__x000A_</S>
	<S S="Error">   n_x000D__x000A_</S>
	<S S="Error"> _x000D__x000A_</S>
	<S S="warning">Log in Warning Stream</S>
	<S S="verbose">Log in Verbose Stream</S>
	<S S="debug">Log in Debug Stream</S>
</Objs>
```

we then extract all the "Error" stream messages to return in our `stderr` stream in the ui _without_ XML, which contains the script we passed followed by:

```
 : Log in Error Stream
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorExcep 
   tion
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorExceptio 
   n
```

(it's still not as pretty as one might like, but it's a faithful extraction of the error stream from the xml (and yes, the crazy line breaks are part of what the stream returns, see the corresponding xml and note we've filtered out the `_x000D__x000A_`) -- expect there are powershell parameters one could set to get better info)

File: camp/camp-brooklyn/src/test/java/org/apache/brooklyn/camp/brooklyn/WindowsYamlLiveTest.java [hunk 1]
```diff
@@ -18,6 +18,10 @@
  */
 package org.apache.brooklyn.camp.brooklyn;
 
+import com.sun.net.httpserver.Authenticator.Success;
```
Comment (grkvlt) [inline camp/camp-brooklyn/src/test/java/org/apache/brooklyn/camp/brooklyn/WindowsYamlLiveTest.java:21]: Really?

Comment (ahgittin) [reply camp/camp-brooklyn/src/test/java/org/apache/brooklyn/camp/brooklyn/WindowsYamlLiveTest.java]: good spot!  oops

**Comment by ahgittin** [comment]
I discovered that the output and host streams were NOT part of the stderr xml returned from winrm.  So made more sense to extract the official powershell streams where possible, and have some conveniences but be clearer what they are.  The PR now does the following:

* stdout = winrm stdout (output/success and host streams, not XML)
* winrm_xml_stderr = winrm stderr (often XML; objects and streams for error, warning, debug, verbose; not output/success or host; as the XML returned by winrm with whitespace formatting)
* stderr = winrm stderr, streams text with XML stripped (streams for error, warning, debug, verbose; not output/success or host, as readable text)
* winrm_error = error stream extracted from winrm_xml_stderr
* winrm_warning = warning stream extracted from winrm_xml_stderr
* winrm_debug = debug stream extracted from winrm_xml_stderr
* winrm_verbose = verbose stream extracted from winrm_xml_stderr

**Comment by ahgittin** [comment]
This is how it looks in practice, for the above script:

image

**Review by duncangrant — APPROVED** [review]
LGTM thanks @ahgittin

**Pull request merged**